app-frontend/src/app/components/common/feedbackModal/feedbackModal.html

```diff
@@ -1,29 +1,52 @@
 <div class="modal-header">
-  <button type="button" class="close" aria-label="Close modal" ng-click="$ctrl.dismiss()">
-    <i aria-hidden="true" class="icon-cross">
-      <span class="sr-only">Close modal</span>
-    </i>
-  </button>
-  <h4 class="modal-title">{{$ctrl.resolve.title}}</h4>
-  <p>{{$ctrl.resolve.subtitle}}</p>
+    <button type="button" class="close" aria-label="Close modal" ng-click="$ctrl.dismiss()">
+        <i aria-hidden="true" class="icon-cross">
+            <span class="sr-only">Close modal</span>
+        </i>
+    </button>
+    <h4 class="modal-title">{{ $ctrl.resolve.title }}</h4>
+    <p>{{ $ctrl.resolve.subtitle }}</p>
 </div>
 <div class="modal-body">
-  <div class="modal-inner-container small text-center">
-      <span class="modal-icon"
-            ng-class="$ctrl.resolve.feedbackIconType">
-        <i ng-class="$ctrl.resolve.feedbackIcon"></i>
-      </span>
-      <div ng-bind-html="$ctrl.resolve.content"></div>
-      <button type="button"
-              class="btn"
-              ng-class="$ctrl.resolve.feedbackBtnType"
-              ng-click="$ctrl.close()">
-        {{$ctrl.resolve.feedbackBtnText}}
-      </button>
-  </div>
+    <div class="modal-inner-container small text-center">
+        <span
+            class="modal-icon"
+            ng-class="$ctrl.resolve.feedbackIconType"
+            ng-if="$ctrl.resolve.feedbackIconType && $ctrl.resolve.feedbackIcon"
+        >
+            <i ng-class="$ctrl.resolve.feedbackIcon"></i>
+        </span>
+        <div ng-bind-html="$ctrl.resolve.content"></div>
+        <button
+            type="button"
+            class="btn"
+            ng-if="$ctrl.resolve.feede"
+            ng-class="$ctrl.resolve.feedbackBtnType && $ctrl.resolve.feedbackBtnText"
+            ng-click="$ctrl.close()"
+        >
+            {{ $ctrl.resolve.feedbackBtnText }}
+        </button>
+    </div>
 </div>
 <div class="modal-footer">
-  <div class="footer-section left">
-    <button type="button" class="btn" ng-click="$ctrl.dismiss()">{{$ctrl.resolve.cancelText}}</button>
-  </div>
+    <div class="footer-section left">
+        <button
+            type="button"
+            class="btn"
+            ng-click="$ctrl.dismiss()"
+            ng-if="$ctrl.resolve.cancelText"
+        >
+            {{ $ctrl.resolve.cancelText }}
+        </button>
+    </div>
+    <div class="footer-section right">
+        <button
+            type="button"
+            class="btn btn-primary"
+            ng-if="$ctrl.resolve.confirmText"
+            ng-click="$ctrl.resolve.onConfirm ? $ctrl.resolve.onConfirm() : $ctrl.dismiss()"
```

Comment:
generally a confirm action should call close() to resolve the modal promise, since calling dismiss() will reject it.

Reply:
Good catch